lang/test/org/partiql/lang/planner/transforms/optimizations/FilterScanToKeyLookupTests.kt

```diff
@@ -0,0 +1,334 @@
+        // todo: scour the code looking for more cases.
+        override fun getParameters() = listOf(
+            // single-field primary key (field-reference on LHS)
+            TestCase(
+                "SELECT * FROM $TABLE_WITH_1_FIELD_PK AS f WHERE f.id = 42",
+                PartiqlPhysical.build {
+                    filter(
+                        DEFAULT_IMPL,
+                        // NOTE: this useless filter will be removed in production by [RemoveUselessFiltersPass].
+                        litTrue(),
+                        project(
+                            impl(FAKE_GET_BY_KEY_PROJECT_OPERATOR_NAME, listOf(ionSymbol(TABLE_WITH_1_FIELD_PK_UUID))),
+                            varDecl(0),
+                            list(litInt(42)),
+                        )
+                    )
+                }
+            ),
+            // single-field primary key (table is list instead of bag)
+            TestCase(
+                "SELECT * FROM $TABLE_OF_TYPE_LIST AS f WHERE f.id = 42",
+                PartiqlPhysical.build {
+                    filter(
+                        DEFAULT_IMPL,
+                        // NOTE: this useless filter will be removed in production by [RemoveUselessFiltersPass].
+                        litTrue(),
+                        project(
+                            impl(FAKE_GET_BY_KEY_PROJECT_OPERATOR_NAME, listOf(ionSymbol(TABLE_OF_TYPE_LIST_UUID))),
+                            varDecl(0),
+                            list(litInt(42)),
+                        )
+                    )
+                }
+            ),
+            // single-field primary key (field-reference on RHS)
+            TestCase(
+                "SELECT * FROM $TABLE_WITH_1_FIELD_PK AS f WHERE 42 = f.id",
+                PartiqlPhysical.build {
+                    filter(
+                        DEFAULT_IMPL,
+                        // NOTE: this useless filter will be removed in production by [RemoveUselessFiltersPass].
+                        litTrue(),
+                        project(
+                            impl(FAKE_GET_BY_KEY_PROJECT_OPERATOR_NAME, listOf(ionSymbol(TABLE_WITH_1_FIELD_PK_UUID))),
+                            varDecl(0),
+                            list(litInt(42)),
+                        )
+                    )
+                }
+            ),
+            // compound primary key. note: WHERE predicate lists fields in different order than
+            // specified in primary key, but they still appear in the correct order in the primary key constructor.
+            TestCase(
+                "SELECT * FROM $TABLE_WITH_3_FIELD_PK AS f WHERE f.marketplaceId = 43 AND f.fulfillmentCenterId = 44 AND f.customerId = 42",
+                PartiqlPhysical.build {
+                    filter(
+                        DEFAULT_IMPL,
+                        // NOTE: this useless `and` expr and filter will be removed in production by
```

Comment:
Perhaps it's coming in a future PR, but would be good to have a test showing the end-to-end behavior with all the optimization passes to show the bigger picture of using the three optimization passes defined in this PR.

Reply:
Interesting, I originally had the `RemoveUselessAnds` and `RemoveUselessFilters` passes added to the `planner` built below, but I ended up removing them because it was obfuscating the actual output of this pass and made development a bit confusing a couple of times.  Instead, I made separate tests for those `RemoveUseless*` passes.